File: Codeforces_Problems/615 Div 3/Main.java
import java.util.*;

public class Main
{
    static Scanner sc = new Scanner(System.in);

    void solve()
    {
        ArrayList<Long>arr = new ArrayList<>();

        long a,b,c,n;
        a = sc.nextLong();
        b = sc.nextLong();
        c = sc.nextLong();
        n = sc.nextLong();

        arr.add(a);
        arr.add(b);
        arr.add(c);

        Collections.sort(arr);

        //System.out.println(arr);

        long res = arr.get(2)*2 - arr.get(1) - arr.get(0);
        n = n - res;

        if(n < 0 || (n % 3) != 0)
            System.out.println("NO");
        else
            System.out.println("YES");
    }
    public static void main(String[] args) 
    {  
        int T = sc.nextInt();

        Main ob = new Main();

        while(T-- > 0)
            ob.solve();
    }
}

File: Codeforces_Problems/615 Div 3/Main.cpp
#include<bits/stdc++.h>
using namespace std;
#define ll long long
#define pb push_back

void solve()
{
    int N;
    cin>>N;

    vector<pair<int,int>>V(N);

    for(int i = 0; i < N; i++)
        cin>>V[i].first>>V[i].second;

    sort(V.begin(),V.end());

    pair<int,int>cur = make_pair(0,0);

    string res = "";

    bool flag = true;

    for(int i = 0; i < N; i++)
    {
        int R = V[i].first - cur.first;
        int U = V[i].second - cur.second;

        if(R < 0 || U < 0)
        {
            cout<<"NO"<<endl;
            flag = false;
            break;
        }

        res += string(R,'R');
        res += string(U,'U');
        cur = V[i];
    } 

    if(flag)
    {
        cout<<"YES"<<endl;
        cout<<res<<endl;
    }
}   

int main()
{
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    int T;
    cin>>T;

    while(T--)
    {
        //cout<<"Hello"<<endl;
        solve();
    }

    return 0;
}